src/sentry/relay/config/experimental.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import time
+from collections.abc import MutableMapping
+from typing import Any, Protocol
+
+import sentry_sdk
+
+logger = logging.getLogger(__name__)
+
+
+class TimeoutException(BaseException):
+    _elapsed: float
+    _timeout: float
+
+    def __init__(self, elapsed: float, timeout: float, *args: object) -> None:
+        super().__init__(*args)
+        self._elapsed = elapsed
+        self._timeout = timeout
+
+
+class BuildTimeChecker:
```

Comment:
mini nit: This type is independent of building anything, so 
```suggestion
class TimeChecker:
```